### PR TITLE
fix: Presentation of SD-JWT are not enveloped anymore in a VP JWT

### LINKED ISSUE #243: Remove kid from verifiable presentation
In the Verifiable Presentation JWS, the `kid` must be removed as we don't have a `kid` in the `cnf.jwk` claim. 

image

### LINKED ISSUE #299: [SD-JWT] Using version SD-JWT 08 instead of SD-JWT 02
The last version of the **[EUDI Wallet](https://italia.github.io/eudi-wallet-it-docs/versione-corrente/en/)** uses the [SD-JWT 02](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-02.html#section-5.2) that supports only disclosures for object property (name-value pairs).

From the [SD-JWT 05+](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-05.html#name-creating-disclosures), disclosures are created differently depending on whether a claim is an object property (key-value pair) or an array element or a nested data structures built from these.

Let us consider the case of the Italian Mobile Driver’s License (mDL):

```json
{
    "cognome": "SOMINI",
    "nome": "FABOI",
    "dataNascita": "1976-03-31",
    "luogoNascita": "THUN (CH )",
    "dataRilascio": "2018-03-29",
    "rilasciataDa": "MIT-UCO",
    "dataScadenza": "2023-08-11",
    "numeroPatente": "U11B00188C",
    "categorie": "A B C D",
    "paeseRilascio": "IT",
    "dettaglioCategorie": [
        {
            "categoria": "A",
            "dataRilascio": "2001-05-10",
            "dataScadenza": "2023-08-11",
            "annotazioniRestrizioni": null
        },
        {
            "categoria": "B",
            "dataRilascio": "1995-01-25",
            "dataScadenza": "2023-08-11",
            "annotazioniRestrizioni": null
        },
        {
            "categoria": "C",
            "dataRilascio": "1996-10-24",
            "dataScadenza": "2023-08-11",
            "annotazioniRestrizioni": null
        },
        {
            "categoria": "D",
            "dataRilascio": "1999-09-15",
            "dataScadenza": "2023-08-11",
            "annotazioniRestrizioni": null
        }
    ],
    "annotazioniRestrizioni": "71 RM7222747K",
    "fotografia": "/9j/4AAQSkZJRgABAQEAyADIAAD/4Qq0RXhpZgAATU0AKgAAAAgAB..."
}
```

Using the **SD-JWT 02**, for the array type property _dettaglioCategorie_, the Issuer MUST creates a flat structure, i.e., all items in the array can only be disclosed in full:

```json
["yA8DdC5SPhA-Ot8nkqRTOA","dettaglioCategorie",[{"annotazioniRestrizioni":null,"categoria":"A","dataRilascio":"2001-05-10","dataScadenza":"2023-08-11"},{"annotazioniRestrizioni":null,"categoria":"B","dataRilascio":"1995-01-25","dataScadenza":"2023-08-11"},{"annotazioniRestrizioni":null,"categoria":"C","dataRilascio":"1996-10-24","dataScadenza":"2023-08-11"},{"annotazioniRestrizioni":null,"categoria":"D","dataRilascio":"1999-09-15","dataScadenza":"2023-08-11"}]]
```

Using the **SD-JWT 05+**, for the array type property _dettaglioCategorie_, the Issuer disclosures each element of the array  individually as described in [Section 5.2.2](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-05.html#disclosures_for_array_elements):

```json
["KWj0ArjU7dsZTwtxJ0sTLw","dettaglioCategorie",[{"...":"5U8J-NLoffwarueQiPeBrwunf3p2qNaadEjKF-OQKzQ"},{"...":"XCFU6ov1n7FPd4QHwTYXBjORqVPRGoYc3Qf8biwjYws"},{"...":"55Ge-Usvam7UzM3pnTL_AjkOUD6C-OzvJIUG1MyDjhU"},{"...":"OGra44m0P40zLrtzqQaQ1BO-_-mawFsL7VZZtCJ22Yg"}]]
```
```json
["31ZwbKDNINo69n8kufwpPQ",{"annotazioniRestrizioni":null,"categoria":"A","dataRilascio":"2001-05-10","dataScadenza":"2023-08-11"}]
```
```json
["72x8vxIkShaQCUiSrCudgw",{"annotazioniRestrizioni":null,"categoria":"B","dataRilascio":"1995-01-25","dataScadenza":"2023-08-11"}]
```
```json
["aRW1FjAcjnMdXp9mBghhog",{"annotazioniRestrizioni":null,"categoria":"C","dataRilascio":"1996-10-24","dataScadenza":"2023-08-11"}]
```
```json
["51AlDWOhuVqI_fJaokBwLA",{"annotazioniRestrizioni":null,"categoria":"D","dataRilascio":"1999-09-15","dataScadenza":"2023-08-11"}]
```

For improve the process of disclosures, we suggest using the version SD-JWT 08 instead of SD-JWT 02.

### PR DESCRIPTION
**Pull request by peppelinux**
This PR removes a deprecable approach we have used during the Wallet PoC where the presented credential, included in the vp_token array, was enveloped in a VP JWT.

This PR clarifies the methods for presenting individual credentials, referring to their specifications.

however, at this stage, clarifications and examples are provided exclusively for SD-JWT and the section dedicated to the presentation of mdoc cbor is To Be Defined.

To be decided whether to enrich this PR with the mdoc cbor part or whether to include this in future PRs.

### CODE REVIEW

File: docs/en/pid-eaa-data-model.rst [hunk 1]
```diff
@@ -60,16 +62,16 @@ The JOSE header contains the following mandatory parameters:
...
-    - Signature Algorithm. 
+    - REQUIRED. Signature Algorithm. 
     - `[RFC7515, Section 4.1.1] <https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1>`_.
   * - **kid**
-    - Unique identifier of the public key. 
+    - REQUIRED. Unique identifier of the public key. 
     - `[RFC7515, Section 4.1.8] <https://datatracker.ietf.org/doc/html/rfc7516.html#section-4.1.8>`_.
   * - **trust_chain**
-    - JSON array containing the trust chain that proves the reliability of the issuer of the JWT. 
+    - Optional. JSON array containing the trust chain that proves the reliability of the issuer of the JWT.
```
Comment (fmarino-ipzs) [inline docs/en/pid-eaa-data-model.rst]:
```suggestion
    - OPTIONAL. JSON array containing the trust chain that proves the reliability of the issuer of the JWT. 
```

**Review by fmarino-ipzs — APPROVED** [review]

**Comment by fmarino-ipzs** [comment]
@peppelinux are you sure that this PR close the issue #299 ? We probably have to align also the data model section.

File: docs/en/pid-eaa-data-model.rst [hunk 2]
```diff
@@ -36,13 +36,15 @@ Each digest value ensures the integrity of, and maps to, the respective Disclosu
...
+The Disclosures are provided to the Holder together with the SD-JWT in the *Combined Format for Issuance* that MUST be an ordered series of base64url-encoded values, each separated from the next by a single tilde ('~') character as follows:
 
 .. code-block::
 
   <Issuer-Signed-JWT>~<Disclosure 1>~<Disclosure 2>~...~<Disclosure N>
 
-See `[SD-JWT VC] <https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-02.html>`_ and `[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-04>`__ for more details. 
+See `[SD-JWT VC] <https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-02.html>`_ and
+`[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-04>`__
+for additional details.
```
Comment (hevelius) [inline docs/en/pid-eaa-data-model.rst]: Why we not always refer to the same version of sd-jwt?

File: docs/en/remote-flow.rst [hunk 3]
```diff
@@ -444,45 +444,62 @@ Where the following parameters are used:
...
-     "jti": "3978344f-8596-4c3a-a978-8fcaba3903c5",
-     "aud": "https://relying-party.example.org/response_uri",
-     "iat": 1541493724,
-     "exp": 1573029723,
-     "nonce": "2c128e4d-fc91-4cd3-86b8-18bdea0988cb",
-     "vp": "<Issuer-Signed-JWT>~<Disclosure 1>~<Disclosure 2>~...~<Disclosure N>"
-   }
+SD-JWT defines how an Holder can present a Credential to a Verifier proving the legitimate possession
+of the Credential. For doing this the Holder MUST include the ``KB-JWT`` in the SD-JWT,
+by appending the ``KB-JWT`` at the end of the of the SD-JWT, as represented in the example below:
```
Comment (hevelius) [inline docs/en/remote-flow.rst:455]: As commented above the `KB-JWT` is in the version 5 or above of sd-jwt spec. So we have refers three different versions of sd-jwt spec here. My concern is reporting different versions of the specification. Is it a safe approach?

File: docs/en/pid-eaa-data-model.rst (hunk 2 above)
Comment (peppelinux) [reply docs/en/pid-eaa-data-model.rst]: they are two linked specs, one sd-JWT and its child SD-JWT-VC

File: docs/en/pid-eaa-data-model.rst [hunk 4]
```diff
@@ -36,13 +36,15 @@ Each digest value ensures the integrity of, and maps to, the respective Disclosu
...
-The Disclosures are sent to the Holder together with the SD-JWT in the *Combined Format for Issuance* that MUST be an ordered series of base64url-encoded values, each separated from the next by a single tilde ('~') character as follows:
+The Disclosures are provided to the Holder together with the SD-JWT in the *Combined Format for Issuance* that MUST be an ordered series of base64url-encoded values, each separated from the next by a single tilde ('~') character as follows:
 
 .. code-block::
 
   <Issuer-Signed-JWT>~<Disclosure 1>~<Disclosure 2>~...~<Disclosure N>
 
-See `[SD-JWT VC] <https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-02.html>`_ and `[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-04>`__ for more details. 
+See `[SD-JWT VC] <https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-02.html>`_ and
+`[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-04>`__
```
Comment (peppelinux) [inline docs/en/pid-eaa-data-model.rst]:
```suggestion
`[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-07>`__
```

File: docs/en/pid-eaa-data-model.rst [hunk 5]
```diff
@@ -60,16 +62,16 @@ The JOSE header contains the following mandatory parameters:
...
+    - REQUIRED. Signature Algorithm. 
     - `[RFC7515, Section 4.1.1] <https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1>`_.
   * - **kid**
-    - Unique identifier of the public key. 
+    - REQUIRED. Unique identifier of the public key. 
     - `[RFC7515, Section 4.1.8] <https://datatracker.ietf.org/doc/html/rfc7516.html#section-4.1.8>`_.
   * - **trust_chain**
-    - JSON array containing the trust chain that proves the reliability of the issuer of the JWT. 
+    - OPTIONAL. JSON array containing the trust chain that proves the reliability of the issuer of the JWT. 
     - `[OIDC-FED, Section 3.2.1] <https://openid.net/specs/openid-connect-federation-1_0.html#name-trust-chain-header-paramete>`_.
```
Comment (peppelinux) [inline docs/en/pid-eaa-data-model.rst]:
```suggestion
    - `[OIDC-FED, Section 3.2.1] <https://openid.net/specs/openid-federation-1_0.html#name-trust-chain-header-paramete>`_.
```

File: docs/en/remote-flow.rst (hunk 3 above)
Comment (peppelinux) [reply docs/en/remote-flow.rst:455]: SD-JWT aligned with 07

I suggest to provide suggestion to speed up the revision phase. The problem statement alone is unfortunately  not enough

File: docs/en/pid-eaa-data-model.rst [hunk 6]
```diff
@@ -36,13 +36,15 @@ Each digest value ensures the integrity of, and maps to, the respective Disclosu
...
-The Disclosures are sent to the Holder together with the SD-JWT in the *Combined Format for Issuance* that MUST be an ordered series of base64url-encoded values, each separated from the next by a single tilde ('~') character as follows:
+The Disclosures are provided to the Holder together with the SD-JWT in the *Combined Format for Issuance* that MUST be an ordered series of base64url-encoded values, each separated from the next by a single tilde ('~') character as follows:
 
 .. code-block::
 
   <Issuer-Signed-JWT>~<Disclosure 1>~<Disclosure 2>~...~<Disclosure N>
 
-See `[SD-JWT VC] <https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-02.html>`_ and `[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-04>`__ for more details. 
+See `[SD-JWT VC] <https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-02.html>`_ and
+`[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-07>`__
```
Comment (peppelinux) [inline docs/en/pid-eaa-data-model.rst]:
```suggestion
`[SD-JWT] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-10>`__
```

File: docs/en/remote-flow.rst [hunk 7]
```diff
@@ -444,45 +444,62 @@ Where the following parameters are used:
...
-    - Unix timestamp beyond which this presentation will no longer be considered valid.
+    - REQUIRED. The value of this claim MUST be the time at which the Key Binding JWT was issued, using the syntax defined in [RFC7519].
   * - **aud**
-    - Audience of the VP, corresponding to the ``response_uri`` within the Authorization request issued by the Relying Party.
+    - REQUIRED. The intended receiver of the Key Binding JWT. How the value is represented is up to the protocol used and out of scope of this specification.
   * - **nonce**
-    - The nonce value provided by the Relying Party within the Authorization Request.
+    - REQUIRED. Ensures the freshness of the signature. The value type of this claim MUST be a string. How this value is obtained is up to the protocol used and out of scope of this specification.
+  * - **sd_hash**
+    - REQUIRED. The base64url-encoded hash digest over the Issuer-signed JWT and the selected Disclosures as defined below.
```
Comment (peppelinux) [inline docs/en/remote-flow.rst]:
```suggestion
    - REQUIRED. The base64url-encoded hash digest over the Issuer-signed JWT and the selected disclosures as defined below.
```

**Review by grausof — APPROVED** [review]

File: docs/en/remote-flow.rst [hunk 8]
```diff
@@ -444,45 +444,62 @@ Where the following parameters are used:
...
-    - The nonce value provided by the Relying Party within the Authorization Request.
+    - REQUIRED. Ensures the freshness of the signature. The value type of this claim MUST be a string. How this value is obtained is up to the protocol used and out of scope of this specification.
+  * - **sd_hash**
+    - REQUIRED. The base64url-encoded hash digest over the Issuer-signed JWT and the selected disclosures as defined below.
+
+
+MDOC-CBOR Presentation
+----------------------
+
+TBD.
```
Comment (peppelinux) [inline docs/en/remote-flow.rst:502]:
```suggestion
MDOC-CBOR Presentation
----------------------

TBD.
```

To be defined.

Comment (peppelinux) [reply docs/en/remote-flow.rst:502]: moved to issue.

**Pull request merged**